mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -244,15 +340,18 @@ class Completions(
         val values = FilenameCompletions.contribute(filename, td)
         (values, true)
       case (lit @ Literal(Constant(_: String))) :: _ =>
-        val completions = InterpolatorCompletions.contribute(
-          pos,
-          completionPos,
-          indexedContext,
-          lit,
-          path,
-          this,
-          config.isCompletionSnippetsEnabled(),
-        )
+        val completions = InterpolatorCompletions
+          .contribute(
+            pos,
+            completionPos,
+            indexedContext,
+            lit,
+            path,
+            this,
+            config.isCompletionSnippetsEnabled(),
+          )
+          .filterInteresting(false)
```

Comment:
Why do we need to `filterInteresting` here? Metals will call `filterInteresting` later, right?
`filterInteresting` is a bit heavy operation even if we don't invoke `enrich`, so we shouldn't call it multiple times.

Reply:
Unfortunately, `filterInteresting` is not applied on any result from the advanced completions. So in case of interpolate completion, this was generating results like `.eq(object: Object)` which should have been filtered.

Reply:
I wanted to apply filterInteresting on the results of `advancedCompletions` too but did not do that just in case there was a good reason to not write it like that, initially. Do you think it would be a good idea to explore that avenue?

Reply:
Yes, I think we should do that (actually Scala2 does), but we can do this in another PR :+1:

Reply:
Let me check and if no test breaks as a result, then I would do it here. but if the tests behave weirdly, I would postpone it to the next PR.

Reply:
I filtered all the advanced completions and no test failed as a result. So do I keep it this way for consistency with what we have for Scala 2?

Reply:
it turned out that filtering the advanced completions was a huge performance toll. So I guess that was not a good idea.

Reply:
I want to filter advanced completion not because I want to make it consistent with Scala2, but more like consistent across all advanced completions (I feel it's wired if we filter only StringInterpolation and not others).
I don't think it makes so much performance deterioration 🤔 but it's ok, we can take a look later :+1:

Reply:
well, basically because of the interruption exception issue, it is very difficult to have a confident opinion on the performance impact. I was as well surprised that this was causing a performance hog, considering the very small size of the advanced completions. But let's solve it later.